pkg/storage/testdata/mvcc_histories/del_with_exclusion_ts

```diff
@@ -0,0 +1,78 @@
+# Write at a timestamp above the exclusion timestamp results in a write exclusion error.
+run ok
+put k=a v=v1 ts=2
+----
+>> at end:
+data: "a"/2.000000000,0 -> /BYTES/v1
+
+run error
+del k=a ts=3 expect-exclusion-since=1
+----
+>> at end:
+data: "a"/2.000000000,0 -> /BYTES/v1
+error: (*kvpb.ExclusionViolationError:) write exclusion on key "a" expected since 1.000000000,0 but found more recent write at 2.000000000,0
+
+run ok
+clear_range k=a end=z
+----
+>> at end:
+<no data>
+
+# Intent at a timestamp above the exclusion timestamp from a different transaction still results in a LockConflictError.
```

Comment:
Note that this is in some sense a contention handling policy decision.  The LockConflictError doesn't escape KV and rather this will result in us waiting on transaction that slipped because of the lock loss. If it commits, we'll end up raising an exclusion violation error.